File: web/app/js/handler/lancamentoProgramadoTransferenciaHandler.js
'use strict';

app.factory('LancamentoProgramadoTransferenciaHandler', function() {
    
    var handleList = function(lancamentoProgramadoTransferenciaList) { 
        return _.map(lancamentoProgramadoTransferenciaList, function(lancamentoProgramadoTransferencia) {
            return handle(lancamentoProgramadoTransferencia);
        });
    };
    
    var handle = function(lancamentoProgramadoTransferencia) {  
        var lancamentoProgramadoTransferenciaHandle = {};
        lancamentoProgramadoTransferenciaHandle.idLancamentoProgramadoTransferencia = getId(lancamentoProgramadoTransferencia);            
        lancamentoProgramadoTransferenciaHandle.lancamentoProgramadoOrigem = getLancamentoProgramadoOrigem(lancamentoProgramadoTransferencia);  
        lancamentoProgramadoTransferenciaHandle.lancamentoProgramadoDestino = getLancamentoProgramadoDestino(lancamentoProgramadoTransferencia);  
        lancamentoProgramadoTransferenciaHandle.numero = getNumero(lancamentoProgramadoTransferencia); 
        lancamentoProgramadoTransferenciaHandle.documento = getDocumento(lancamentoProgramadoTransferencia); 
        lancamentoProgramadoTransferenciaHandle.formaPagamento = getFormaPagamento(lancamentoProgramadoTransferencia); 
        lancamentoProgramadoTransferenciaHandle.frequencia = getFrequencia(lancamentoProgramadoTransferencia);
        lancamentoProgramadoTransferenciaHandle.dataEmissao = getDataEmissao(lancamentoProgramadoTransferencia);
        lancamentoProgramadoTransferenciaHandle.dataCompetenciaOrigem = getDataCompetenciaOrigem(lancamentoProgramadoTransferencia);
        lancamentoProgramadoTransferenciaHandle.dataCompetenciaDestino = getDataCompetenciaDestino(lancamentoProgramadoTransferencia);
        lancamentoProgramadoTransferenciaHandle.dataVencimentoOrigem = getDataVencimentoOrigem(lancamentoProgramadoTransferencia);
        lancamentoProgramadoTransferenciaHandle.dataVencimentoDestino = getDataVencimentoDestino(lancamentoProgramadoTransferencia);
        lancamentoProgramadoTransferenciaHandle.valor = getValor(lancamentoProgramadoTransferencia);
        lancamentoProgramadoTransferenciaHandle.historico = getHistorico(lancamentoProgramadoTransferencia);
        lancamentoProgramadoTransferenciaHandle.observacao = getObservacao(lancamentoProgramadoTransferencia);
        lancamentoProgramadoTransferenciaHandle.usuario = getUsuario(lancamentoProgramadoTransferencia);
        return lancamentoProgramadoTransferenciaHandle;
    };
    
    var getId = function(lancamentoProgramadoTransferencia) {
        return lancamentoProgramadoTransferencia.idLancamentoProgramadoTransferencia || null;
    };
    
    var getLancamentoProgramadoOrigem = function(lancamentoProgramadoTransferencia) {
        if(!lancamentoProgramadoTransferencia.lancamentoProgramadoOrigem) return lancamentoProgramadoTransferencia.lancamentoProgramadoOrigem;
        return { idLancamentoProgramado: lancamentoProgramadoTransferencia.lancamentoProgramadoOrigem.idLancamentoProgramado }; 
    };
    
    var getLancamentoProgramadoDestino = function(lancamentoProgramadoTransferencia) {
        if(!lancamentoProgramadoTransferencia.lancamentoProgramadoDestino) return lancamentoProgramadoTransferencia.lancamentoProgramadoDestino;
        return { idLancamentoProgramado: lancamentoProgramadoTransferencia.lancamentoProgramadoDestino.idLancamentoProgramado }; 
    };
    
    var getNumero = function(lancamentoProgramadoTransferencia) {
        return lancamentoProgramadoTransferencia.numero || null;
    };
    
    var getDocumento = function(lancamentoProgramadoTransferencia) {
        if(!lancamentoProgramadoTransferencia.documento) return lancamentoProgramadoTransferencia.documento;
        return { idTipoDocumento: lancamentoProgramadoTransferencia.documento.idTipoDocumento };
    };
    
    var getFormaPagamento = function(lancamentoProgramadoTransferencia) {
        if(!lancamentoProgramadoTransferencia.formaPagamento) return lancamentoProgramadoTransferencia.formaPagamento;
        return { idTipoFormaPagamento: lancamentoProgramadoTransferencia.formaPagamento.idTipoFormaPagamento };
    };
    
    var getFrequencia = function(lancamentoProgramadoTransferencia) {
        if(!lancamentoProgramadoTransferencia.frequencia) return lancamentoProgramadoTransferencia.frequencia;
        return lancamentoProgramadoTransferencia.frequencia.id;
    };
    
    var getDataEmissao = function(lancamentoProgramadoTransferencia) {
        return lancamentoProgramadoTransferencia.dataEmissao || moment(); 
    };
    
    var getDataCompetenciaOrigem = function(lancamentoProgramadoTransferencia) {
        return lancamentoProgramadoTransferencia.dataCompetenciaOrigem || moment(); 
    };
    
    var getDataCompetenciaDestino = function(lancamentoProgramadoTransferencia) {
        return lancamentoProgramadoTransferencia.dataCompetenciaDestino || moment(); 
    };
    
    var getDataVencimentoOrigem = function(lancamentoProgramadoTransferencia) {
        return lancamentoProgramadoTransferencia.dataVencimentoOrigem || moment(); 
    };
    
    var getDataVencimentoDestino = function(lancamentoProgramadoTransferencia) {
        return lancamentoProgramadoTransferencia.dataVencimentoDestino || moment(); 
    };
    
    var getValor = function(lancamentoProgramadoTransferencia) {
        return lancamentoProgramadoTransferencia.valor || 0;
    };
    
    var getHistorico = function(lancamentoProgramadoTransferencia) {
        return lancamentoProgramadoTransferencia.historico || null;
    };
    
    var getObservacao = function(lancamentoProgramadoTransferencia) {
        return lancamentoProgramadoTransferencia.observacao || null;
    };
    
    var getUsuario = function(lancamentoProgramadoTransferencia) {
        return lancamentoProgramadoTransferencia.usuario || null;
    };

    return {
        handle: handle,
        handleList: handleList
    }

});

File: src/java/com/portalpostal/dao/handler/LancamentoProgramadoTransferenciaHandler.java
package com.portalpostal.dao.handler;

import com.portalpostal.model.LancamentoProgramado;
import com.portalpostal.model.LancamentoProgramadoTransferencia;
import com.portalpostal.model.TipoDocumento;
import com.portalpostal.model.TipoFormaPagamento;
import com.portalpostal.model.dd.TipoFrequencia;
import java.sql.ResultSet;
import java.sql.SQLException;
import org.sql2o.ResultSetHandler;

public class LancamentoProgramadoTransferenciaHandler extends GenericHandler implements ResultSetHandler<LancamentoProgramadoTransferencia> {
        
    public LancamentoProgramadoTransferenciaHandler() {
        super("lancamento_transferencia_programado");
    }
    
    public LancamentoProgramadoTransferenciaHandler(String table) {
        super(table);
    }

    public LancamentoProgramadoTransferencia handle(ResultSet result) throws SQLException {
        LancamentoProgramadoTransferencia lancamento = new LancamentoProgramadoTransferencia();
        lancamento.setIdLancamentoProgramadoTransferencia(getInt(result, "idLancamentoProgramadoTransferencia"));
        lancamento.setLancamentoProgramadoOrigem(getLancamentoProgramadoOrigem(result));
        lancamento.setLancamentoProgramadoDestino(getLancamentoProgramadoDestino(result));
        lancamento.setNumero(getString(result, "numero"));
        lancamento.setDocumento(getTipoDocumento(result));
        lancamento.setFormaPagamento(getTipoFormaPagamento(result));
        lancamento.setFrequencia(TipoFrequencia.values()[getInt(result, "frequencia")]);
        lancamento.setDataCompetencia(getDate(result, "dataCompetencia"));
        lancamento.setDataEmissao(getDate(result, "dataEmissao"));
        lancamento.setDataVencimento(getDate(result, "dataVencimento"));
        lancamento.setValor(getDouble(result, "valor"));
        lancamento.setHistorico(getString(result, "historico"));
        lancamento.setObservacao(getString(result, "observacao"));
        lancamento.setUsuario(getString(result, "usuario"));
        return lancamento;
    }
    
    private LancamentoProgramado getLancamentoProgramadoOrigem(ResultSet result) throws SQLException {
        if(!existColumn(result, "lancamentoProgramadoOrigem.idLancamentoProgramado")) return null;
        return new LancamentoProgramadoHandler("lancamentoProgramadoOrigem").handle(result); 
    }
    
    private LancamentoProgramado getLancamentoProgramadoDestino(ResultSet result) throws SQLException {
        if(!existColumn(result, "lancamentoProgramadoDestino.idLancamentoProgramado")) return null;
        return new LancamentoProgramadoHandler("lancamentoProgramadoDestino").handle(result); 
    }
    
    private TipoDocumento getTipoDocumento(ResultSet result) throws SQLException {
        if(!existColumn(result, "tipo_documento.idTipoDocumento")) return null;
        return new TipoDocumentoHandler().handle(result); 
    }
    
    private TipoFormaPagamento getTipoFormaPagamento(ResultSet result) throws SQLException {
        if(!existColumn(result, "tipo_forma_pagamento.idTipoFormaPagamento")) return null;
        return new TipoFormaPagamentoHandler().handle(result); 
    }
    
}
